training/SYMBOL_TABLE_OVERVIEW.md

```diff
@@ -0,0 +1,170 @@
+
+`st.findAll(regex string)` is currently only used in `register/attach` functionality as it was specifically designed to handle cases of `attaching` to multiple-component data objects that don't have a Chapel equivalent, such as `Series` or `DataFrame`, easily. It is used when you have a partial name that can map to multiple entries in a set. An example of when this is used is when `Categorical` objects are registered to the symbol table. These objects register 4 separate entries, one for each piece of the `Categorical` needed to rebuild it on the client. 
+
+If you have a set of data with entries `dataset_entry_0`, `dataset_entry_1`, ... `dataset_entry_n` and you don't know how many entries are in this set, this function makes it easy to find all of the entries. By passing the portion of the name these entries have in common, `dataset_entry_`, followed by regex notation for "Any number, 1 or more times", `/d+`. We can then get a list of all the matching names in the symbol table and operate on all the dataset pieces.
+
+```chapel
+  var entryList = st.findAll("dataset_entry_/d+");
+  // entryList: [`dataset_entry_0`, `dataset_entry_1`, 'dataset_entry_2']
+```
+
+You can then loop through the returned array of names to access the data and perform operations on each entry in the set.
+
+<a id="accessSym"></a>
+## Accessing Symbol Table Entry Data
+
+Depending on your functionality requirements, there are a couple different ways to access a symbol table entry.
+
+### Passing the Entry to the Client
+
+If you need to return an entry to python, the `SymTab.attrib(name)` method will return the entry's metadata, which is required by the arkouda client to reference the entry when operations are sent to the server to be performed on the object. You can then use the return message to `attach` to the entry and view the data as you would with any other Python object.
+
+It is important to note that on the client after `attaching` to a Symbol Table Entry, the client does not store the data or perform computations on the data. The client is only storing the metadata and any computations on the object are performed on the server.
+
+More information on passing data between the client and server can be found [here](https://github.com/Bears-R-Us/arkouda/blob/master/training/MESSAGING_OVERVIEW.md).
+
+### Working with Entries in Chapel
+
+When working with a basic `SymEntry` entry type, you can easily access the data values within the entry by accessing the property `a`. To access the domain information, access the property `aD`. For example, assume we have a SymTab entry variable called `entry`.
+
+```chapel
+  var values = entry.a;
+  var domain = entry.aD;

```

Comment:
we should name our `domain` variable since that is a reserved keyword in chapel

Reply:
Renamed to `entryDomain`